### PR TITLE
Handle get-export-status responses

### PR DESCRIPTION
**Pull request by nguyen102**
Description of changes:
* Added code to handle the different responses for an export status request.

**Related PRs**
* https://github.com/awslabs/fhir-works-on-aws-interface/pull/14
* https://github.com/awslabs/fhir-works-on-aws-authz-rbac/pull/8

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/router/routes/exportRoute.ts [hunk 1]
```diff
@@ -59,17 +59,43 @@ export default class ExportRoute {
 
         // Export Job Status
         this.router.get(
-            '/\\$export/:jobId',
+            '/\\$export/:requesterUserId/:jobId',
```
Comment (rsmayda) [inline src/router/routes/exportRoute.ts]: Hmm is this expected part of the spec?

Comment (nguyen102) [reply src/router/routes/exportRoute.ts]: The FHIR spec didn't specify what the job status URL be, so this is up for us to decide. I'm including this so we can use it to authorized the user. We get the user's `userId` from their `accessToken` and compare it to the `requesterUserId` here to confirm that they are the same user. Given that they are authorized to get this job's status.

Comment (rsmayda) [reply src/router/routes/exportRoute.ts]: Let's talk offline on a good solution

File: src/router/handlers/exportHandler.ts [hunk 2]
```diff
@@ -16,11 +16,11 @@ export default class ExportHandler {
         return this.bulkDataAccess.initiateExport(initiateExportRequest);
```
Comment (rsmayda) [inline src/router/handlers/exportHandler.ts:26]: So looking at the code for bundles it looks like AuthZ still happens in the routing layer: https://github.com/awslabs/fhir-works-on-aws-routing/blob/mainline/src/router/bundle/bundleHandler.ts#L127

Thoughts on mimicing this approach?

**Review by rsmayda — APPROVED** [review]

File: src/router/handlers/exportHandler.ts [hunk 3]
```diff
@@ -34,13 +35,19 @@ export default class ExportHandler {
     async cancelExport(jobId: string, requesterUserId: string): Promise<void> {
         const jobDetails = await this.bulkDataAccess.getExportStatus(jobId);
         await this.checkIfRequesterHasAccessToJob(jobDetails, requesterUserId);
+        if (['completed', 'failed'].includes(jobDetails.jobStatus)) {
```
Comment (rsmayda) [inline src/router/handlers/exportHandler.ts:38]: I think the only time you can cancel the request is when it is in 'pending' right?

Comment (nguyen102) [reply src/router/handlers/exportHandler.ts:38]: The different type of `jobStatus` are

` 'completed' | 'failed' | 'in-progress' | 'canceled' | 'canceling'`

If the job is currently in `'in-progress' | 'canceled' | 'canceling'`, we don't want to throw an error for the cancel request. We can just proceed as is.

Comment (rsmayda) [reply src/router/handlers/exportHandler.ts:38]: What will the service do when the job in canceled gets asked to cancel again? Basically could we return from here?

**Pull request merged**